analytics/src/main/java/com/segment/analytics/internal/StopMessage.java

```diff
@@ -0,0 +1,61 @@
+package com.segment.analytics.internal;
+
+import com.segment.analytics.messages.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Date;
+import java.util.Map;
+
+class StopMessage implements Message {
+  static final StopMessage STOP = new StopMessage();
+
+  private StopMessage() {}
+
+  @Nonnull
+  @Override
+  public Type type() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nonnull
+  @Override
+  public String messageId() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nonnull
+  @Override
+  public Date timestamp() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nullable
+  @Override
+  public Map<String, ?> context() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nullable
+  @Override
+  public String anonymousId() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nullable
+  @Override
+  public String userId() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Nullable
+  @Override
+  public Map<String, Object> integrations() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String toString() {
+    return "FlushMessage{}";
+  }
+}
```

Comment:
Also, I think it's about time to create a SignalMessage super-class to signal something that's not a message itself but an action within the queue.
That way we can only change two attributes of the child-class(toString and the static one) and have it ready

Reply:
Maybe even make it an enum?